### PR TITLE
Fixes to methane Tech Note

### PR DESCRIPTION
**Pull request by samsrabin**
### Description of changes

Various minor fixes to the methane section of the Tech Note:
- Added missing citation links
- Added missing equation references
- Corrected superscript syntax
- Corrected units in a table
- Corrected use of $\phi$ to $\varphi$

Equation references still missing:
- [x] "The gaseous concentration is calculated with Henry’s law as described in equation ."
- [x] "… with the same representation as Equation but with the gas diffusivity of oxygen."
- [x] "The solution to equation is solved in several sequential steps: …"

### Specific notes

Contributors other than yourself, if any: @jinmuluo

CTSM Issues Fixed: Item 3 in #1990.

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any: Rendered HTML looks good.

### CODE REVIEW

**Comment by wwieder** [comment]
@samsrabin I think that: 

1.  _the gaseous concentration is calculated with Henry’s law as described in equation **(2.25.7)**_, since C_{e} is the expected equilibrium gaseous partial pressure.
2. _O2 can also diffuse in from the atmosphere to the soil layer via the reverse of the same pathway, with the same representation as Equation **(2.25.8)** but with the gas diffusivity of oxygen_ because A is the diffusive transport (of CH4) through aerenchyma.

**Comment by samsrabin** [comment]
Thanks, @wwieder!

File: doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst [hunk 1]
```diff
@@ -426,7 +426,7 @@ molecular free-air diffusion coefficients (:math:`{D}_{0}`
 .. table:: Temperature dependence of aqueous and gaseous diffusion coefficients for CH\ :sub:`4` and O\ :sub:`2`
 
  +----------------------------------------------------------+----------------------------------------------------------+--------------------------------------------------------+
- | :math:`{D}_{0}` (m\ :sup:`2` s\ :sup:`-1`)               | CH\ :sub:`4`                                             | O\ :sub:`2`                                            |
+ | :math:`{D}_{0}` (cm\ :sup:`2` s\ :sup:`-1`)              | CH\ :sub:`4`                                             | O\ :sub:`2`                                            |
```
Comment (wwieder) [inline doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst:429]: I'm assuming you checked these units with the code?

File: doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst [hunk 2]
```diff
@@ -259,8 +259,8 @@ aqueous CH\ :sub:`4` concentration, and *p* is pressure.
 The local pressure is calculated as the sum of the ambient pressure,
 water pressure down to the local depth, and pressure from surface
 ponding (if applicable). When the CH\ :sub:`4` partial pressure
-exceeds 15% of the local pressure (Baird et al. 2004; Strack et al.
-2006; Wania et al. 2010), bubbling occurs to remove CH\ :sub:`4`
+exceeds 15% of the local pressure (:ref:`Baird et al. 2004<Bairdetal2004>`; :ref:`Strack et al.
+2006<Stracketal2006>`; :ref:`Wania et al. 2010<Waniaetal2010>`), bubbling occurs to remove CH\ :sub:`4`
```
Comment (wwieder) [inline doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst:263]: Thanks for adding these reference links!

File: doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst [hunk 3]
```diff
@@ -286,14 +286,14 @@ The diffusive transport through aerenchyma (*A*, mol m\ :sup:`-2` s\ :sup:`-1`)
...
 concentration at depth *z* (m); :math:`r_{L}`  is the ratio of root
 length to depth; *p* is the porosity (-); *T* is specific aerenchyma
-area (m:sup:`2` m\ :sup:`-2`); :math:`{r}_{a}` is the
+area (m\ :sup:`2` m\ :sup:`-2`); :math:`{r}_{a}` is the
 aerodynamic resistance between the surface and the atmospheric reference
-height (s m:sup:`-1`); and :math:`\rho _{r}`  is the rooting
+height (s m\ :sup:`-1`); and :math:`\rho _{r}`  is the rooting
 density as a function of depth (-). The gaseous concentration is
-calculated with Henry’s law as described in equation .
+calculated with Henry’s law as described in equation :eq:`24.7`.
```
Comment (wwieder) [inline doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst:296]: I wonder why all of these were missing in the first place?  Thanks for adding these references.

**Review by wwieder — APPROVED** [review]
Thanks for making these updates, @samsrabin

File: doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst (hunk 1 above)
Comment (samsrabin) [reply doc/source/tech_note/Methane/CLM50_Tech_Note_Methane.rst:429]: Yep! The constants are set here, where the correct units are given in the comment:
https://github.com/samsrabin/CTSM/blob/7ae1372c0c39747ba71ae399cd2c5964e72f917d/src/main/clm_varcon.F90#L244-L247

When they're used, they're multiplied by 1e-4 to convert `D0` from cm2/s to m2/s:
https://github.com/samsrabin/CTSM/blob/7ae1372c0c39747ba71ae399cd2c5964e72f917d/src/soilbiogeochem/SoilBiogeochemNitrifDenitrifMod.F90#L400

However, it looks like the table is missing a digit. Will add.

**Pull request merged**